### PR TITLE
WIP: Add a silent option to environment constructor

### PR DESCRIPTION
**Pull request by simonbowly**
Possible fix for the stray output you are seeing with 9.5 as noted in #437

1. Printing the license owners's email address: this can only be silenced by creating an empty environment, setting outputflag to zero, then starting the environment.
2. Stray printing of NonConvex parameters. This would also be silenced by this strategy, however setting OutputFlag first should have also achieved this. Is there something different done by full_bridge_optimizer that might be interfering with the Silent setting? It seems to be only these tests which print parameters.

### CODE REVIEW

File: src/MOI_wrapper/MOI_wrapper.jl [hunk 1]
```diff
@@ -97,10 +97,15 @@ mutable struct Env
     finalize_called::Bool
     attached_models::Int
 
-    function Env()
+    function Env(::MOI.Silent, flag::Bool)
```
Comment (odow) [inline src/MOI_wrapper/MOI_wrapper.jl:100]: Why not just `Env(; output_flag::Int = 1)`?

**Review by odow — COMMENTED** [review]
CI failed because you pushed to your `simonbowly` fork. If you want to run the CI, you can push to the `jump-dev` remote:

```
git add remote jump-dev https://github.com/jump-dev/Gurobi.jl.git
git push -u jump-dev silent-env
```

**Comment by odow** [comment]
Ah, I may have found the issue: 
https://github.com/jump-dev/Gurobi.jl/blob/1648d6e316835b492020bac9ee4e22bf02e5102b/src/MOI_wrapper/MOI_wrapper.jl#L382-L388

Looks like we set silent _after_ we change other parameters 😥

**Comment by simonbowly** [comment]
Ah, nice one! And that's called in every constructor?

**Comment by odow** [comment]
If you create a new `Optimizer`, or call` empty!`, yes. Let's just switch the two statements.

**Comment by simonbowly** [comment]
I still get a couple of outputs when running the tests with that change. Are there a few Silent=False settings hanging about?

```
Set parameter OutputFlag to value 1
Set parameter OutputFlag to value 1
Set parameter NonConvex to value 2
Set parameter DualReductions to value 0
Set parameter InfUnbdInfo to value 1
Set parameter QCPDual to value 1
```

**Comment by simonbowly** [comment]
CI test run for just the parameter ordering change is here - https://github.com/simonbowly/Gurobi.jl/runs/5240766098?check_suite_focus=true

This cleans up everything except the licensing messages, second commit is needed to do that.

**Pull request closed without merging**